### PR TITLE
chore:search bar list items redirection to profile page

### PR DESCRIPTION
**Pull request by pushpendrakukreti**
### Features

- Added homepage link to navbar logo 
- Removed search icon from search bar
- Only firstname to show in session dropown
- User won't able to clear search box and there should be a cancel button in search text box
so that user will on search box and clear search result.
- User can search by friend by first and last name.

image

### CODE REVIEW

**Review by priyajoshipj — CHANGES_REQUESTED** [review]
Please find below issue's
1) User won't able to clear search box and there should be a cancel button in search text box
so that user will on search box and clear search result.
2) User can search by friend by first and last name.
3) Friend list UI should be in horizontal order.

**Comment by pushpendrakukreti** [comment]
> Updated following points

> 1. User won't able to clear search box and there should be a cancel button in search text box
>    so that user will on search box and clear search result.
> 2. User can search by friend by first and last name.
> 3. Friend list UI should be in horizontal order.  (will update this point in next PR)
> 4.  Fix backspacing issue in search box, it should be cleared properly.

File: src/scenes/navbar/index.jsx [hunk 1]
```diff
@@ -61,13 +65,23 @@ const Navbar = () => {
...
 
   const navigateHome = () => {
     // 👇️ navigate to /
     navigate('/home');
   };
 
+  const routeChange = (userId) => {
+    let path = `/profile/${userId}`;
+    navigate(path);
+    navigate(0);
```
Comment (priyajoshipj) [inline src/scenes/navbar/index.jsx:82]: why navigate(0) ?

File: src/scenes/navbar/index.jsx [hunk 2]
```diff
@@ -85,7 +99,7 @@ const Navbar = () => {
   //setSearch
   return (
     <FlexBetween padding="1rem 6%" backgroundColor={alt}>
-      <FlexBetween gap="1.75rem">
+      <FlexBetween gap="1.75rem" height={"7vh"}>
```
Comment (priyajoshipj) [inline src/scenes/navbar/index.jsx:102]: try to avoid inline styles use classes instead

File: src/scenes/navbar/index.jsx [hunk 3]
```diff
@@ -114,14 +128,21 @@ const Navbar = () => {
...
               />
+              <Icon>
+                <Clear
+                  className="clear-search"
+                  onClick={() => handlerSearch("")}
+                ></Clear>
+              </Icon>
             </FlexBetween>
           )}
           <List class={searchUserList.length > 0 ? "search-box-container" : ""}>
```
Comment (priyajoshipj) [inline src/scenes/navbar/index.jsx:139]: you can optimized your code like this class={ `${searchUserList.length.length > 0  && "search-box-container" ` }

**Review by priyajoshipj — CHANGES_REQUESTED** [review]

File: src/scenes/navbar/index.jsx [hunk 4]
```diff
@@ -48,14 +53,35 @@ const Navbar = () => {
...
   };
 
+  const routeChange = (userId) => {
+    let path = `/profile/${userId}`;
+    navigate(path);
+    navigate(0);
+  }
+
   const getSearchResult = async (values) => {
     const savedUserResponse = await fetch(
```
Comment (priyajoshipj) [inline src/scenes/navbar/index.jsx:86]: please add try and catch to catch api error

**Review by priyajoshipj — CHANGES_REQUESTED** [review]

File: src/scenes/navbar/index.jsx (hunk 1 above)
Comment (pushpendrakukreti) [reply src/scenes/navbar/index.jsx:82]: When we redirect to any profile, and search for another profile on same page it just updates the id in url not the page.
navigate(0) refreshes the page by taking current url.

**Review by priyajoshipj — APPROVED** [review]

**Pull request merged**